spec/services/protocol_importers/search_protocols_service_spec.rb

```diff
@@ -32,6 +32,40 @@
     end
   end
 
+  context 'when raise api client error' do
+    it 'return network errors' do
+      allow_any_instance_of(ProtocolImporters::ProtocolsIO::V3::ApiClient)
+        .to(receive(:protocol_list)
+        .and_raise(SocketError))
+
+      expect(service_call.errors).to have_key(:socketerror)
+    end
+
+    it 'return api errors' do
+      allow_any_instance_of(ProtocolImporters::ProtocolsIO::V3::ApiClient)
+        .to(receive(:protocol_list)
+        .and_raise(ProtocolImporters::ProtocolsIO::V3::ApiErrors::MissingOrEmptyParametersError.new('1', 'Missing Or Empty Parameters Error')))
+
+      expect(service_call.errors).to have_key(:"protocolimporters::protocolsio::v3::apierrors::missingoremptyparameterserror")
```

Comment:
We should also test other errors defined in the `ApiClient` - in the `ApiClientSpec`.